cardano-node/docs/GenesisGeneration.md

```diff
@@ -0,0 +1,159 @@
+
+### ftsSeed
+
+The key `ftsSeed` contains seed value required for Follow-the-Satoshi mechanism (hex-encoded).
+
+### protocolConsts
+The key `protocolConsts` contains basic protocol constants:
+
+- `k` - security parameter from the paper,
+- `protocolMagic` - protocol magic section, described fully below:
+- `pm` - protocol magic number.
+- `requiresNetworkMagic` - either "NMMustBeNothing" or - "NMMustBeJust",
+- `vssMaxTTL` - VSS certificates maximum timeout to live (number of epochs),
+- `vssMinTTL` - VSS certificates minimum timeout to live (number of epochs).
+
+### protocolMagic
+The key `protocolMagic` defines the protocol magic number. When the protocol magic is changed, all signatures become invalid. This is used to distinguish different networks.
+
+- `pm` - is the protocol magic number, is included in serialized blocks and headers, and is part of signed data.
+- `requiresNetworkMagic` - will be either `NMMustBeNothing` or `NMMustBeJust`.
+The `protocolMagic` value can either be an object with the two fields described above, or just a plain integer. In the latter case, "requiresNetworkMagic" will take the default value of `NMMustBeJust`.
+
+The "requiresNetworkMagic" setting forms part of the genesis data. However it is configured in the core section.
+
+### heavyDelegation
+The key `heavyDelegation` contains an information about heavyweight delegation:
+
+- `cert` - delegation certificate.
+- `delegatePk` - delegate's public key.
+- `issuerPk` - stakeholder's (issuer's) public key.
+- `omega` - index of epoch the block PSK is announced in, it is needed for replay attack prevention, can be set to arbitrary value in genesis.
+
+Keys in `heavyDelegation` dictionary are StakeholderIds of certificates' issuers. They must be consistent with issuerPk values.
+
+### startTime
+The key `startTime` is the timestamp of the 0-th slot. Ideally it should be few seconds later than the cluster actually starts. If it's significantly later, nodes won't be doing anything for a while. If it's slightly before the actual starts, some slots will be missed, but it shouldn't be critical as long as less than k slots are missed.
+
+### vssCerts
+
+The key `vssCerts` contains VSS certificates:
+
+- `expiryEpoch` - index of epoch until which (inclusive) the certificate is valid.
+- `signature` - signature of certificate.
+- `signingKey` - key used for signing.
+- `vssKey` - VSS public key.
+
+Keys in `vssCerts` dictionary are StakeholderIds of certificates' issuers. They must be consistent with signingKey values.
+
+## Balances and stakes
+In genesis data there are two fields which determine balances: `avvmDistr` and `nonAvvmBalances`
+
+The former is a map from AVVM addresses to balances
+The latter is a map from Byron addresses to balances.
+AVVM addresses can be easily converted to Byron addresses (redeem type), so obtaining a final map from Byron addresses to balances is trivial.
```

Comment:
Should these be prefixed with `- ` for consistency with the above sections? Not a big deal, just wondering if they're different somehow.

Reply:
Done